src/ray/core_worker/core_worker.h

```diff
@@ -858,6 +1010,7 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   /// Whether we are shutting down and not running further tasks.
   bool exiting_ = false;
 
+  friend class CoreWorkerProcess;
```

Comment:
Let's please try to avoid having this friend class, it leads to some really ugly abstractions. I think it's only necessary because of the few fields accessed during shutdown - can we just move that logic to a method instead?

Reply:
Done.